### PR TITLE
[PFOE-8] Nowy branch żeby posprzątać bałagan

### PR DESCRIPTION
**Pull request by olafmyszak**

### CODE REVIEW

File: backend/WebApi/WebApi/Controllers/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public IActionResult Login(string login, string password)
+    {
+        var loggedUser = CheckLogin(login, password);
+
+        if (loggedUser != null)
+        {
+            return Ok(loggedUser);
+        }
+
+        return BadRequest();
```
Comment (MCMisha) [inline backend/WebApi/WebApi/Controllers/UserController.cs:33]: Lepiej zrobić tak:
`return NotFound();`

File: backend/WebApi/WebApi/Tests/UserServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
+using NUnit.Framework;
+using WebApi.Models;
+using WebApi.Services;
+
+namespace  WebApi.Tests;
+
+[TestFixture]
+public class UserServiceTests
```
Comment (MCMisha) [inline backend/WebApi/WebApi/Tests/UserServiceTests.cs:8]: Zrób najpierw testy do repozitorium

File: backend/WebApi/WebApi/Tests/UserServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        var config = new ConfigurationBuilder()
+            .AddJsonFile("appsettings.json")
+            .AddEnvironmentVariables()
+            .Build();
+        return config;
+    }
+
+    [Test]
+    public void Login_ValidUser_ReturnsUser()
```
[the remark this answers is not in the document]
Comment (olafmyszak) [reply backend/WebApi/WebApi/Tests/UserServiceTests.cs:20]: Ale unit testy powinny być niezależne od bazy danych i reszty systemu

[the remark this answers is not in the document]
Comment (MCMisha) [reply backend/WebApi/WebApi/Tests/UserServiceTests.cs:20]: Wtedy musisz przerobić na coś takiego:
`using NUnit.Framework;
using Moq;
using Microsoft.Extensions.Logging;
using Microsoft.Extensions.Configuration;
using Microsoft.AspNetCore.Mvc;
using WebApi.Models;
using WebApi.Controllers;
using WebApi.Services;

namespace WebApi.Tests.Controllers
{
    [TestFixture]
    public class UserControllerTests
    {
        private UserController _userController;
        private Mock<UserService> _mockUserService;
        private Mock<HashService> _mockHashService;
        private Mock<ILogger<UserController>> _mockLogger;

        [SetUp]
        public void Setup()
        {
            _mockUserService = new Mock<UserService>(MockBehavior.Strict, Mock.Of<IConfiguration>());
            _mockHashService = new Mock<HashService>(MockBehavior.Strict);
            _mockLogger = new Mock<ILogger<UserController>>();

            _userController = new UserController(_mockLogger.Object, Mock.Of<IConfiguration>());
            _userController.LoginChecker = () => null;
        }

        [Test]
        public void Login_Returns_OkResult_When_LoginIsSuccessful()
        {
            // Arrange
            string login = "user";
            string password = "password";
            var user = new User { Login = login, Password = "hashedPassword" };
            _mockUserService.Setup(service => service.Login(login, "hashedPassword")).Returns(user);
            _mockHashService.Setup(service => service.GetSha256Hash(password)).Returns("hashedPassword");

            // Act
            var result = _userController.Login(login, password);

            // Assert
            Assert.IsInstanceOf<OkObjectResult>(result);
            var okResult = (OkObjectResult)result;
            Assert.AreEqual(user, okResult.Value);
        }

        [Test]
        public void Login_Returns_BadRequest_When_LoginFails()
        {
            // Arrange
            string login = "user";
            string password = "password";
            _mockUserService.Setup(service => service.Login(login, "hashedPassword")).Returns((User)null);
            _mockHashService.Setup(service => service.GetSha256Hash(password)).Returns("hashedPassword");

            // Act
            var result = _userController.Login(login, password);

            // Assert
            Assert.IsInstanceOf<BadRequestResult>(result);
        }

        [Test]
        public void CheckEmail_Returns_OkResult_When_EmailExists()
        {
            // Arrange
            string email = "test@example.com";
            _mockUserService.Setup(service => service.CheckEmail(email)).Returns(true);

            // Act
            var result = _userController.CheckEmail(email);

            // Assert
            Assert.IsInstanceOf<OkObjectResult>(result);
            var okResult = (OkObjectResult)result;
            Assert.AreEqual(true, okResult.Value);
        }

        [Test]
        public void CheckEmail_Returns_OkResult_When_EmailDoesNotExist()
        {
            // Arrange
            string email = "test@example.com";
            _mockUserService.Setup(service => service.CheckEmail(email)).Returns(false);

            // Act
            var result = _userController.CheckEmail(email);

            // Assert
            Assert.IsInstanceOf<OkObjectResult>(result);
            var okResult = (OkObjectResult)result;
            Assert.AreEqual(false, okResult.Value);
        }
    }
}
`

File: backend/WebApi/WebApi/Controllers/UserController.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        return NotFound();
+    }
+
+    [HttpGet("checkEmail")]
+    public IActionResult CheckEmail(string email)
+    {
+        return Ok(_userService.CheckEmail(email));
+    }
```
Comment (MCMisha) [inline backend/WebApi/WebApi/Controllers/UserController.cs:38]: Zrób jeszcze
`[HttpGet("checkLogin")]`

**Review by MCMisha — APPROVED** [review]

**Pull request merged**